cli/src/main/java/org/jboss/as/cli/impl/aesh/cmd/HelpCommand.java

```diff
@@ -178,6 +190,13 @@ public CommandResult execute(CLICommandInvocation commandInvocation) throws Comm
             }
         }
 
+        // Since the introduction of Prospero as the tool to update, the "patch" command only makes sense in a domain
+        // mode to patch legacy host controllers. Therefore, we allow the Help Command to progress with "patch" command
+        // only in domain mode.
+        if ("patch".equals(mainCommand)  && !ctx.isDomainMode()) {
+            throw new CommandException("The command is not available in the current context.");
```

Comment:
I misunderstood the first revision from @bstansberry , he was talking about the message thrown by the exception and I thought he was talking about the code comments.

The Exception message is printed directly in the CLI output, so it can also be used to give more context. 

So we could use it to show more information. The difficulties are what sort of message we want to add here. I could suggest the following:

"
The 'patch' command is only available in a Domain Mode to patch legacy Host Controllers but is deactivated for standalone or disconnected sessions. See 'help installer' to get more information about how to patch your server.
"

Maybe we could review the "installer" help page to also explain when you have to use the Prospero tool and when the `installer`. Notice the `installer` command is not available on embedded Server / Host controllers, basically because we need to restart a running server to apply the update.

I see both options, allowing "help patch" to progress and adding a more descriptive text there sounds also reasonable.

Reply:
@jfdenise Could you add your opinion about this one?

I guess, I am just waiting to know whether we want to fix this by describing the information in the "patch" help page without removing the availability of "help patch" or by removing the ability of the "help patch" and showing the above message in the exception

Reply:
@yersan , I feel that I prefer the help doc to be updated. The command exists but is just enabled in some cases. The help doc could explain.

Reply:
Thanks @jfdenise , I've updated the PR accordingly

Reply:
@jfdenise Could you review again? thanks!